cli/azd/pkg/project/project_config.go

```diff
@@ -101,13 +105,7 @@ func (pc *ProjectConfig) GetProject(ctx *context.Context, env *environment.Envir
 		Services: make([]*Service, 0),
 	}
 
-	// This sets the current template within the go context
-	// The context is then used when the AzCli is instantiated to set the correct user agent
-	if project.Metadata != nil && strings.TrimSpace(project.Metadata.Template) != "" {
-		*ctx = telemetry.ContextWithTemplate(*ctx, project.Metadata.Template)
-	}
```

Comment:
Can you fix the comment that it is only used by `AzCli` but leave this around? The template name set in context is registered as a telemetry baggage -- which means any telemetry spans that use the context end up including it as a property, so it is being used by our telemetry events outside of `AzCli`.

I probably misunderstood this when we spoke offline about this, as you mentioned about the concern that `AzCli` was dependent on this behavior.

Reply:
Eventually, this should land into a place where it automatically gets injected during `wire` as we parse the project config. The current place where this is, is definitely weird, and only happens as part of `Project` construction and not `ProjectConfig`.

Reply:
Hmm, now that I'm looking at it, the code has changed significantly since the initial release and I'm not sure if this working quite as expected as before

Reply:
I reverted this part of the diff and changed the comment to read:

```go
	// This sets the current template within the go context
	// The template name set in context is registered as a telemetry baggage so any telemetry spans that use
	// the context end up including it as a property.
```

Reply:
Per our Teams conversation we belive this is actually dead code now, so I removed it yet again and in doing so was able to change the type of the context argument from *context.Context to just context.Context (since we no longer add anything to the context) so I folded that in.